crates/ra_syntax/test_data/parser/err/0036_partial_use.rast

```diff
@@ -20,32 +20,22 @@ SOURCE_FILE@0..37
             PATH_SEGMENT@17..22
               NAME_REF@17..22
                 IDENT@17..22 "Error"
-        ERROR@22..23
-          SEMICOLON@22..23 ";"
-        WHITESPACE@23..24 "\n"
-        ERROR@24..27
-          USE_KW@24..27 "use"
-        WHITESPACE@27..28 " "
-        USE_TREE@28..35
-          PATH@28..35
-            PATH@28..31
-              PATH_SEGMENT@28..31
-                NAME_REF@28..31
-                  IDENT@28..31 "std"
-            COLON2@31..33 "::"
-            PATH_SEGMENT@33..35
-              NAME_REF@33..35
-                IDENT@33..35 "io"
-        ERROR@35..36
-          SEMICOLON@35..36 ";"
+    SEMICOLON@22..23 ";"
+  WHITESPACE@23..24 "\n"
```

Comment:
This is an example of the new behavior being superior on a pre-existing test case; we emit many fewer spurious errors, and manage to recover in time to capture the `use` tree on line 2 properly.